File: solutions/zig/rotational-cipher/1/rotational_cipher.zig
const std = @import("std");
const mem = std.mem;

pub fn rotate(allocator: mem.Allocator, text: []const u8, shiftKey: u5) mem.Allocator.Error![]u8 {
    var result = try allocator.alloc(u8, text.len);
    errdefer allocator.free(result);

    for (text, 0..) |char, i| {
        result[i] = if (char >= 'a' and char <= 'z')
            'a' + (@as(u8, char) - 'a' + @as(u8, shiftKey)) % 26
        else if (char >= 'A' and char <= 'Z')
            'A' + (@as(u8, char) - 'A' + @as(u8, shiftKey)) % 26
        else
            char;
    }

    return result;
}

File: solutions/typescript/rotational-cipher/1/rotational-cipher.ts
export function rotate(text: string, shiftKey: number): string {
    let result = '';
    for (const char of text) {
        const charCode = char.charCodeAt(0);
        if (charCode >= 65 && charCode <= 90) {
            result += String.fromCharCode(
                ((charCode - 65 + shiftKey) % 26) + 65,
            );
        } else if (charCode >= 97 && charCode <= 122) {
            result += String.fromCharCode(
                ((charCode - 97 + shiftKey) % 26) + 97,
            );
        } else {
            result += char;
        }
    }
    return result;
}
